File: examples/delphi/my-app/src/Utils/Logger.pas
unit Utils.Logger;

interface

uses
  System.SysUtils,
  System.Classes;

type
  TLogger = class
  private
    FLogFile: string;
    procedure WriteToFile(const AMessage: string);
  public
    constructor Create;
    destructor Destroy; override;
    
    procedure Log(const AMessage: string);
    procedure LogError(const AMessage: string);
    procedure LogWarning(const AMessage: string);
    procedure LogDebug(const AMessage: string);
  end;

implementation

constructor TLogger.Create;
begin
  inherited;
  FLogFile := 'app.log';
end;

destructor TLogger.Destroy;
begin
  inherited;
end;

procedure TLogger.WriteToFile(const AMessage: string);
begin
  // File writing logic here
end;

procedure TLogger.Log(const AMessage: string);
var
  Timestamp: string;
begin
  Timestamp := FormatDateTime('yyyy-mm-dd hh:nn:ss', Now);
  WriteToFile(Format('[%s] INFO: %s', [Timestamp, AMessage]));
end;

procedure TLogger.LogError(const AMessage: string);
var
  Timestamp: string;
begin
  Timestamp := FormatDateTime('yyyy-mm-dd hh:nn:ss', Now);
  WriteToFile(Format('[%s] ERROR: %s', [Timestamp, AMessage]));
end;

procedure TLogger.LogWarning(const AMessage: string);
var
  Timestamp: string;
begin
  Timestamp := FormatDateTime('yyyy-mm-dd hh:nn:ss', Now);
  WriteToFile(Format('[%s] WARNING: %s', [Timestamp, AMessage]));
end;

procedure TLogger.LogDebug(const AMessage: string);
var
  Timestamp: string;
begin
  Timestamp := FormatDateTime('yyyy-mm-dd hh:nn:ss', Now);
  WriteToFile(Format('[%s] DEBUG: %s', [Timestamp, AMessage]));
end;

end.

File: examples/kotlin/my-app/src/main/kotlin/com/example/utils/Logger.kt
package com.example.utils

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object Logger {
    private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
    
    fun info(message: String) {
        log("INFO", message)
    }
    
    fun debug(message: String) {
        log("DEBUG", message)
    }
    
    fun warn(message: String) {
        log("WARN", message)
    }
    
    fun error(message: String) {
        log("ERROR", message)
    }
    
    private fun log(level: String, message: String) {
        val timestamp = LocalDateTime.now().format(formatter)
        println("[$timestamp] [$level] $message")
    }
}
